### ISSUE TITLE
Docker error for kong 0.9.9 : Error starting userland proxy

### ISSUE THREAD

**Issue by mohammedharoon** [Mashape/kong#2926]
I am using below steps to run cassandra 2.2 . It is running fine, but when I am trying to run kong, docker is throwing error.
1.
sudo docker run -d --name kong-database \
                -p 9042:9042 \
                cassandra:2.2
after running cassandra container wait some time for initialization else host fail error will show up.

2.
sudo docker run --rm \
    --link kong-database:kong-database \
    -e "KONG_DATABASE=cassandra" \
    -e "KONG_PG_HOST=kong-database" \
    -e "KONG_CASSANDRA_CONTACT_POINTS=kong-database" \
    kong:0.9.9 kong migrations up

3.
sudo docker run -d --name kong \
    --link kong-database:kong-database \
    -e "KONG_DATABASE=cassandra" \
    -e "KONG_CASSANDRA_CONTACT_POINTS=kong-database" \
    -e "KONG_PG_HOST=kong-database" \
    -p 8000:8000 \
    -p 8443:8443 \
    -p 8001:8001 \
    -p 7946:7946 \
    kong:0.9.9

and I am getting the error as: 
docker: Error response from daemon: driver failed programming external connectivity on endpoint kong (a83d8ca21feef88df3f4dc05d33134aa6428e8b4404bafe2265d0053b67a0fb7): Error starting userland proxy: listen tcp 0.0.0.0:7946: bind: address already in use.

The tcp port is not being used anywhere. Before running the above commands I am using fuser -k 7946/tcp to make sure the port is not being used anywhere.
Please can someone help me resolve this.

**Comment by hutchic** [comment]
0.9.9 is a bit of an older release may I ask why this version?

`kong:0.9.9` didn't have a dedicated migration command. The following worked for me

```
docker run -d --name kong-database -p 9042:9042 cassandra:2.2

docker run -d --name kong \
--link kong-database:kong-database \
-e "KONG_DATABASE=cassandra" \
-e "KONG_CASSANDRA_CONTACT_POINTS=kong-database" \
-e "KONG_PG_HOST=kong-database" \
-p 8000:8000 \
-p 8443:8443 \
-p 8001:8001 \
-p 7946:7946 \
kong:0.9.9

curl -i http://localhost:8001/
HTTP/1.1 200 OK
Date: Tue, 03 Oct 2017 14:46:22 GMT
Content-Type: application/json; charset=utf-8
Transfer-Encoding: chunked
Connection: keep-alive
Access-Control-Allow-Origin: *
Server: kong/0.9.9
```

```
uname -a
Linux ip-172-31-24-121 4.4.0-1013-aws #22-Ubuntu SMP Fri Mar 31 15:41:31 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux

docker --version
Docker version 17.05.0-ce, build 89658be
```

**Comment by mohammedharoon** [comment]
Thanks for the response.  With kong 0.11 the admin API have changed a bit and is not compatible with my application so I am using kong 0.9.9.

I tried your approach again but getting the same error.  For some reason the port 7946 gives the same error "Error starting userland proxy: listen tcp 0.0.0.0:7946: bind: address already in use". I changed the port mapping as -p 9001:7946 and it is working fine as of now. Issue is with the port binding i guess.

Thanks a lot for your time.
